qiskit_nature/problems/second_quantization/lattice/lattices/hyper_cubic_lattice.py

```diff
@@ -0,0 +1,304 @@
+
+        Raises:
+            ValueError: When edge parameter or boundary condition is a tuple,
+                the length of that is not the same as that of size.
+        """
+
+        self.dim = len(size)
+
+        self.size = size
+
+        # edge parameter
+        if isinstance(edge_parameter, (int, float, complex)):
+            edge_parameter = (edge_parameter,) * self.dim
+        elif isinstance(edge_parameter, tuple):
+            if len(edge_parameter) != self.dim:
+                raise ValueError(
+                    "size mismatch, "
+                    f"`edge_parameter`: {len(edge_parameter)}, `size`: {self.dim}."
+                    "The length of `edge_parameter` must be the same as that of size."
+                )
+
+        self.edge_parameter = edge_parameter
+
+        self.onsite_parameter = onsite_parameter
+
+        # boundary condition
+        if isinstance(boundary_condition, BoundaryCondition):
+            boundary_condition = (boundary_condition,) * self.dim
+        elif isinstance(boundary_condition, tuple):
+            if len(boundary_condition) != self.dim:
+                raise ValueError(
+                    "size mismatch, "
+                    f"`boundary_condition`: {len(boundary_condition)}, `size`: {self.dim}."
+                    "The length of `boundary_condition` must be the same as that of size."
+                )
+
+        self.boundary_condition = boundary_condition
+
+        graph = PyGraph(multigraph=False)
+        graph.add_nodes_from(range(np.prod(size)))
+
+        # add edges excluding the boundary edges
+        bulk_edge_list = self._bulk_edges()
+        graph.add_edges_from(bulk_edge_list)
+
+        # add self-loops.
+        self_loop_list = self._self_loops()
+        graph.add_edges_from(self_loop_list)
+
+        # add edges that cross the boundaries
+        boundary_edge_list = self._boundary_edges()
+        graph.add_edges_from(boundary_edge_list)
+
+        # a list of edges that depend on the boundary condition
+        self.boundary_edges = [(edge[0], edge[1]) for edge in boundary_edge_list]
+
+        super().__init__(graph)
+
+        # default position for one and two-dimensional cases.
+        self.pos = self._default_position()
+
+    def draw_without_boundary(
+        self,
+        self_loop: bool = False,
+        style: Optional[LatticeDrawStyle] = None,
+    ):
+        r"""Draw the lattice with no edges between the boundaries.
+
+        Args:
+            self_loop: Draw self-loops in the lattice. Defaults to False.
+            style : Styles for retworkx.visualization.mpl_draw.
+                Please see
+                https://qiskit.org/documentation/retworkx/stubs/retworkx.visualization.mpl_draw.html#retworkx.visualization.mpl_draw
+                for details.
+
+        Returns:
+            A matplotlib figure for the visualization if not running with an
+            interactive backend (like in jupyter) or if ``ax`` is not set.
```

Comment:
I think this function just draws the lattice and does not return anything. It has no return typehint but from what I can see it should be typehinted as None since it does not explicitly return anything.